cpp/periodic_module/periodic_iterate.cpp

```diff
@@ -0,0 +1,306 @@
+#include <fmt/core.h>
+#include <mgp.hpp>
+
+#include "mgclient.hpp"
+
+const char *kProcedurePeriodic = "iterate";
+const char *kArgumentInputQuery = "input_query";
+const char *kArgumentRunningQuery = "running_query";
+const char *kArgumentConfig = "config";
+const char *kConfigKeyBatchSize = "batch_size";
+const char *kBatchInternalName = "__batch";
+const char *kBatchRowInternalName = "__batch_row";
+const char *kReturnSuccess = "success";
+const char *kReturnNumBatches = "number_of_executed_batches";
+
+const char *kMgHost = "MG_HOST";
+const char *kMgPort = "MG_PORT";
+const char *kMgUsername = "MG_USERNAME";
+const char *kMgPassword = "MG_PASSWORD";
+
+const char *kDefaultHost = "localhost";
+const uint16_t kDefaultPort = 7687;
+
+struct ParamNames {
+  std::vector<std::string> node_names;
+  std::vector<std::string> relationship_names;
+  std::vector<std::string> primitive_names;
+};
+
+auto ExtractParamNames(const std::vector<std::string> &columns, const std::vector<mg::Value> &batch_row) {
+  ParamNames res;
+  for (size_t i = 0; i < columns.size(); i++) {
+    if (batch_row[i].type() == mg::Value::Type::Node) {
+      res.node_names.push_back(columns[i]);
+    } else if (batch_row[i].type() == mg::Value::Type::Relationship) {
+      res.relationship_names.push_back(columns[i]);
+    } else {
+      res.primitive_names.push_back(columns[i]);
+    }
+  }
+
+  return res;
+}
+
+auto Join(const std::vector<std::string> &strings, const std::string &delimiter) {
+  if (!strings.size()) {
+    return std::string();
+  }
+
+  auto result = strings[0];
+
+  for (size_t i = 1; i < strings.size(); i++) {
+    result += delimiter + strings[i];
+  }
+
+  return result;
+}
+
+auto GetGraphFirstClassEntityAlias(const std::string &internal_name, const std::string &entity_name) {
+  return fmt::format("{}.{} AS __{}_id", internal_name, entity_name, entity_name);
+}
+
+auto GetPrimitiveEntityAlias(const std::string &internal_name, const std::string &primitive_name) {
+  return fmt::format("{}.{} AS {}", internal_name, primitive_name, primitive_name);
+}
+
+auto ConstructWithStatement(const ParamNames &names) {
+  std::vector<std::string> with_entity_vector;
+  for (const auto &node_name : names.node_names) {
+    with_entity_vector.push_back(GetGraphFirstClassEntityAlias(kBatchRowInternalName, node_name));
```

Comment:
emplace

Reply:
And without `GetGraphFirstClassEntityAlias` as we want to not use ctor and move ctor then